core/src/main/java/org/mini2Dx/core/engine/interpolator/geom/LinearCircleInterpolator.java

```diff
@@ -0,0 +1,27 @@
+package org.mini2Dx.core.engine.interpolator.geom;
+
+import org.mini2Dx.core.engine.interpolator.Interpolator;
+import org.mini2Dx.core.engine.interpolator.Interpolators;
+import org.mini2Dx.core.geom.Circle;
+
+/**
+ * The circle interpolation function
+ */
+public final class LinearCircleInterpolator implements Interpolator<Circle> {
+
+  public static final LinearCircleInterpolator INSTANCE = new LinearCircleInterpolator();
+
+  private LinearCircleInterpolator() {
+  }
+
+  @Override
+  public Circle interpolate(Circle flyweight, Circle initialValue, Circle targetValue, float alpha) {
```

Comment:
This class shows how more abstract classes can use the interpolators recursively; the circle has 3 floats that needs to be interpolated, but we don't want to re-write that interpolation logic everywhere.